File: Python/A_Project/B_Snake_and_Food/FoodItemFactory/food_factory.py
from typing import Sequence

from FoodItemFactory.ConcreteFoodItem.bonus_food import BonusFood
from FoodItemFactory.ConcreteFoodItem.normal_food import NormalFood


class FoodFactory:
    @staticmethod
    def create_food(pos: Sequence[int], food_type: str):
        try:
            row, col = pos[0], pos[1]
        except (TypeError, IndexError):
            raise ValueError("Position must contain at least 2 elements") from None
        if food_type == "bonus":
            return BonusFood(row, col)
        return NormalFood(row, col)

File: Ruby/A_Project/B_Snake_And_Food/FoodItemFactory/food_factory.rb
# frozen_string_literal: true

require_relative "ConcreteFoodItem/bonus_food"
require_relative "ConcreteFoodItem/normal_food"

class FoodFactory
  def self.create_food(pos, food_type)
    row = pos[0]
    col = pos[1]

    raise ArgumentError, "Position must contain at least 2 elements" if row.nil? || col.nil?

    if food_type == "bonus"
      BonusFood.new(row, col)
    else
      NormalFood.new(row, col)
    end
  end
end
